agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/components/AgendaRefreshableList.kt

```diff
@@ -0,0 +1,90 @@
+package com.nibalk.tasky.agenda.presentation.components
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.Spring
+import androidx.compose.animation.core.VectorConverter
+import androidx.compose.animation.core.spring
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.Text
+import androidx.compose.material3.pulltorefresh.PullToRefreshBox
+import androidx.compose.material3.pulltorefresh.PullToRefreshState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.nibalk.tasky.core.presentation.themes.TaskyTheme
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> AgendaRefreshableList(
+    items: List<T>,
+    content: @Composable (T) -> Unit,
+    isRefreshing: Boolean,
+    onRefresh: () -> Unit,
+    modifier: Modifier = Modifier,
+    lazyListState: LazyListState = rememberLazyListState()
+) {
+
+    val pullToRefreshState = remember {
+        object : PullToRefreshState {
+            private val anim = Animatable(0f, Float.VectorConverter)
+
+            override val distanceFraction
+                get() = anim.value
+
+            override suspend fun animateToThreshold() {
+                anim.animateTo(1f, spring(dampingRatio = Spring.DampingRatioHighBouncy))
+            }
+
+            override suspend fun animateToHidden() {
+                anim.animateTo(0f)
+            }
+
+            override suspend fun snapTo(targetValue: Float) {
+                anim.snapTo(targetValue)
+            }
+        }
+    }
+
+    PullToRefreshBox(
+        modifier = modifier,
+        isRefreshing = isRefreshing,
+        onRefresh = onRefresh,
+        state = pullToRefreshState
+    ) {
+        LazyColumn(
+            modifier = Modifier.fillMaxSize(),
+            state = lazyListState,
+        ) {
+            if (!isRefreshing) {
```

Comment:
The normal UX would be to keep on showing the old items until new ones are loaded, right? The load could be slow and the user wouldn't be able to do anything meanwhile